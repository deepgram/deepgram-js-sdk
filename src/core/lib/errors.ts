export class DeepgramError extends Error {
  protected __dgError = true;

  constructor(message: string) {
    super(message);
    this.name = "DeepgramError";
  }
}

export function isDeepgramError(error: unknown): error is DeepgramError {
  return typeof error === "object" && error !== null && "__dgError" in error;
}

export class DeepgramApiError extends DeepgramError {
  status: number;

  constructor(message: string, status: number) {
    super(message);
    this.name = "DeepgramApiError";
    this.status = status;
  }

  toJSON() {
    return {
      name: this.name,
      message: this.message,
      status: this.status,
    };
  }
}

export class DeepgramUnknownError extends DeepgramError {
  originalError: unknown;

  constructor(message: string, originalError: unknown) {
    super(message);
    this.name = "DeepgramUnknownError";
    this.originalError = originalError;
  }
}

export class DeepgramVersionError extends DeepgramError {
  constructor() {
    super(
      `You are attempting to use an old format for a newer SDK version. Read more here: https://dpgr.am/js-v3`
    );

    this.name = "DeepgramVersionError";
  }
}

/**
 * Enhanced WebSocket error that captures additional debugging information
 * including status codes, request IDs, and response headers when available.
 */
export class DeepgramWebSocketError extends DeepgramError {
  originalEvent?: ErrorEvent | Event;
  statusCode?: number;
  requestId?: string;
  responseHeaders?: Record<string, string>;
  url?: string;
  readyState?: number;

  constructor(
    message: string,
    options: {
      originalEvent?: ErrorEvent | Event;
      statusCode?: number;
      requestId?: string;
      responseHeaders?: Record<string, string>;
      url?: string;
      readyState?: number;
    } = {}
  ) {
    super(message);
    this.name = "DeepgramWebSocketError";
    this.originalEvent = options.originalEvent;
    this.statusCode = options.statusCode;
    this.requestId = options.requestId;
    this.responseHeaders = options.responseHeaders;
    this.url = options.url;
    this.readyState = options.readyState;
  }

  toJSON() {
    return {
      name: this.name,
      message: this.message,
      statusCode: this.statusCode,
      requestId: this.requestId,
      responseHeaders: this.responseHeaders,
      url: this.url,
      readyState: this.readyState,
      originalEvent: this.originalEvent
        ? {
            type: this.originalEvent.type,
            timeStamp: this.originalEvent.timeStamp,
          }
        : undefined,
    };
  }
}
