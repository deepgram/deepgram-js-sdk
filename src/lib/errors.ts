export class DeepgramError extends Error {
  protected __dgError = true;

  constructor(message: string) {
    super(message);
    this.name = "DeepgramError";
  }
}

export function isDeepgramError(error: unknown): error is DeepgramError {
  return typeof error === "object" && error !== null && "__dgError" in error;
}

export class DeepgramApiError extends DeepgramError {
  status: number;

  constructor(message: string, status: number) {
    super(message);
    this.name = "DeepgramApiError";
    this.status = status;
  }

  toJSON() {
    return {
      name: this.name,
      message: this.message,
      status: this.status,
    };
  }
}

export class DeepgramUnknownError extends DeepgramError {
  originalError: unknown;

  constructor(message: string, originalError: unknown) {
    super(message);
    this.name = "DeepgramUnknownError";
    this.originalError = originalError;
  }
}

export class DeepgramVersionError extends DeepgramError {
  constructor() {
    super(
      `You are attempting to use an old format for a newer SDK version. Read more here: https://dpgr.am/js-v3`
    );

    this.name = "DeepgramVersionError";
  }
}
