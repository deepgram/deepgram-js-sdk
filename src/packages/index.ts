export { AbstractClient } from "./AbstractClient";
export { AbstractRestfulClient } from "./AbstractRestfulClient";
export { AbstractWsClient } from "./AbstractWsClient";
export { ListenClient } from "./ListenClient";
export { LiveClient } from "./LiveClient";
export { ManageClient } from "./ManageClient";
export { OnPremClient } from "./OnPremClient";
export { PrerecordedClient } from "./PrerecordedClient";
