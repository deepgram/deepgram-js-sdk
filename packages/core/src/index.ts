import { DeepgramClientOptions, IKeyFactory } from "./lib/types/DeepgramClientOptions";
import { DeepgramVersionError } from "./lib/errors";
import DeepgramClient from "./DeepgramClient";

/**
 * This class is deprecated and should not be used. It throws a `DeepgramVersionError` when instantiated.
 *
 * @deprecated
 * @see https://dpgr.am/js-v3
 */
class Deepgram {
  constructor(protected apiKey: string, protected apiUrl?: string, protected requireSSL?: boolean) {
    throw new DeepgramVersionError();
  }
}

/**
 * Creates a new Deepgram client instance.
 *
 * @param {DeepgramClientArgs} args - Arguments to pass to the Deepgram client constructor.
 * @returns A new Deepgram client instance.
 */
function createClient(): DeepgramClient;
function createClient(key?: string | IKeyFactory): DeepgramClient;
function createClient(options?: DeepgramClientOptions): DeepgramClient;
function createClient(key?: string | IKeyFactory, options?: DeepgramClientOptions): DeepgramClient;
function createClient(
  keyOrOptions?: string | IKeyFactory | DeepgramClientOptions,
  options?: DeepgramClientOptions
): DeepgramClient {
  let resolvedOptions: DeepgramClientOptions = {};

  if (typeof keyOrOptions === "string" || typeof keyOrOptions === "function") {
    if (typeof options === "object") {
      resolvedOptions = options;
    }

    resolvedOptions.key = keyOrOptions;
  } else if (typeof keyOrOptions === "object") {
    resolvedOptions = keyOrOptions;
  }

  return new DeepgramClient(resolvedOptions);
}

export { createClient, DeepgramClient, Deepgram };

/**
 * Helpful exports.
 */
export * from "./packages";
export * from "./lib/types";
export * from "./lib/enums";
export * from "./lib/constants";
export * from "./lib/errors";
export * from "./lib/helpers";

/**
 * Captions. These will be tree-shaken if unused.
 *
 * @see https://github.com/deepgram/deepgram-node-captions
 *
 * import/export declarations don't do anything but set up an alias to the
 * exported variable, they do not count as a "use". Given their semantics,
 * they are tracked specially by any bundler and will not adversely affect
 * tree-shaking.
 */
export { webvtt, srt } from "@deepgram/captions";
