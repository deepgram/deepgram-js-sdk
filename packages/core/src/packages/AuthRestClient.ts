import { isDeepgramError } from "../lib/errors";
import type { DeepgramResponse } from "../lib/types/DeepgramResponse";
import type { GrantTokenResponse } from "../lib/types/GrantTokenResponse";
import type { GrantTokenSchema } from "../lib/types/GrantTokenSchema";
import { AbstractRestClient } from "./AbstractRestClient";

export class AuthRestClient extends AbstractRestClient {
  public namespace: string = "auth";

  /**
   * Generates a new temporary token for the Deepgram API.
   * @param options Optional configuration options for the token generation. Includes ttl_seconds to set token expiration.
   * @param endpoint Optional custom endpoint to use for the request. Defaults to ":version/auth/grant".
   * @returns Object containing the result of the request or an error if one occurred. Result will contain access_token and expires_in properties.
   */
  public async grantToken(
    options: GrantTokenSchema = {},
    endpoint = ":version/auth/grant"
  ): Promise<DeepgramResponse<GrantTokenResponse>> {
    try {
      const requestUrl = this.getRequestUrl(endpoint);
      const body = JSON.stringify(options);
      const result: GrantTokenResponse = await this.post(requestUrl, body, {
        headers: { "Content-Type": "application/json" },
      }).then((result) => result.json());

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }
}
