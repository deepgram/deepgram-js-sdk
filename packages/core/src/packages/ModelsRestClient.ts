import { isDeepgramError } from "../lib/errors";
import {
  DeepgramResponse,
  GetModelResponse,
  GetModelsResponse,
  GetModelsSchema,
} from "../lib/types";
import { AbstractRestClient } from "./AbstractRestClient";

/**
 * Represents a REST client for interacting with the Deepgram API.
 *
 * The `ModelsRestClient` class provides methods for interacting with the Deepgram API to retrieve information about available models.
 * @extends AbstractRestClient
 */
export class ModelsRestClient extends AbstractRestClient {
  public namespace: string = "models";

  /**
   * Retrieves a list of all available models.
   *
   * @param endpoint - (optional) The endpoint to request.
   * @returns A promise that resolves with the response from the Deepgram API.
   * @example
   * ```typescript
   * import { createClient } from "@deepgram/sdk";
   *
   * const deepgram = createClient(DEEPGRAM_API_KEY);
   * const { result: models, error } = deepgram.models.getAll();
   *
   * if (error) {
   *   console.error(error);
   * } else {
   *   console.log(models);
   * }
   * ```
   */
  async getAll(
    endpoint = ":version/models",
    options: GetModelsSchema = {}
  ): Promise<DeepgramResponse<GetModelsResponse>> {
    try {
      const requestUrl = this.getRequestUrl(endpoint, {}, options);
      const result: GetModelsResponse = await this.get(requestUrl).then((result) => result.json());

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }

  /**
   * Retrieves information about a specific model.
   *
   * @param modelId - The UUID of the model to retrieve.
   * @param endpoint - (optional) The endpoint to request.
   * @returns A promise that resolves with the response from the Deepgram API.
   * @example
   * ```typescript
   * import { createClient } from "@deepgram/sdk";
   *
   * const deepgram = createClient(DEEPGRAM_API_KEY);
   * const { result: model, error } = deepgram.models.getModel("modelId");
   *
   * if (error) {
   *   console.error(error);
   * } else {
   *   console.log(model);
   * }
   * ```
   */
  async getModel(
    modelId: string,
    endpoint = ":version/models/:modelId"
  ): Promise<DeepgramResponse<GetModelResponse>> {
    try {
      const requestUrl = this.getRequestUrl(endpoint, { modelId });
      const result: GetModelResponse = await this.get(requestUrl).then((result) => result.json());

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }
}
