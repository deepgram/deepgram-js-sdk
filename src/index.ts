import DeepgramClient from "./DeepgramClient";
import { DeepgramVersionError } from "./lib/errors";
import type { DeepgramClientOptions } from "./lib/types";

/**
 * Major version fallback error
 */
class Deepgram {
  constructor(protected apiKey: string, protected apiUrl?: string, protected requireSSL?: boolean) {
    throw new DeepgramVersionError();
  }
}

/**
 * Creates a new Deepgram Client.
 */
const createClient = (apiKey: string, options: DeepgramClientOptions = {}): DeepgramClient => {
  return new DeepgramClient(apiKey, options);
};

export { createClient, DeepgramClient, Deepgram };

/**
 * Helpful exports.
 */
export * from "./packages";
export * from "./lib/types";
export * from "./lib/enums";
export * from "./lib/constants";
export * from "./lib/errors";
export * from "./lib/helpers";

/**
 * Captions. These will be tree-shaken if unused.
 *
 * @see https://github.com/deepgram/deepgram-node-captions
 *
 * import/export declarations don't do anything but set up an alias to the
 * exported variable, they do not count as a "use". Given their semantics,
 * they are tracked specially by any bundler and will not adversely affect
 * tree-shaking.
 */
export { webvtt, srt } from "@deepgram/captions";
