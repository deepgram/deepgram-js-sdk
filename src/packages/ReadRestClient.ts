import { CallbackUrl, isTextSource, isUrlSource } from "../lib/helpers";
import { DeepgramError, isDeepgramError } from "../lib/errors";
import type {
  AnalyzeSchema,
  AsyncAnalyzeResponse,
  DeepgramResponse,
  SyncAnalyzeResponse,
  TextSource,
  UrlSource,
} from "../lib/types";
import { AbstractRestClient } from "./AbstractRestClient";

/**
 * The `ReadRestClient` class extends the `AbstractRestClient` class and provides methods for analyzing audio sources synchronously and asynchronously.
 *
 * The `analyzeUrl` method analyzes a URL-based audio source synchronously, returning a promise that resolves to the analysis response or an error.
 *
 * The `analyzeText` method analyzes a text-based audio source synchronously, returning a promise that resolves to the analysis response or an error.
 *
 * The `analyzeUrlCallback` method analyzes a URL-based audio source asynchronously, returning a promise that resolves to the analysis response or an error.
 *
 * The `analyzeTextCallback` method analyzes a text-based audio source asynchronously, returning a promise that resolves to the analysis response or an error.
 */
export class ReadRestClient extends AbstractRestClient {
  public namespace: string = "read";

  /**
   * Analyzes a URL-based audio source synchronously.
   *
   * @param source - The URL-based audio source to analyze.
   * @param options - Optional analysis options.
   * @param endpoint - The API endpoint to use for the analysis. Defaults to ":version/read".
   * @returns A promise that resolves to the analysis response, or an error if the analysis fails.
   */
  async analyzeUrl(
    source: UrlSource,
    options?: AnalyzeSchema,
    endpoint = ":version/read"
  ): Promise<DeepgramResponse<SyncAnalyzeResponse>> {
    try {
      let body;

      if (isUrlSource(source)) {
        body = JSON.stringify(source);
      } else {
        throw new DeepgramError("Unknown source type");
      }

      if (options !== undefined && "callback" in options) {
        throw new DeepgramError(
          "Callback cannot be provided as an option to a synchronous transcription. Use `analyzeUrlCallback` or `analyzeTextCallback` instead."
        );
      }

      const requestUrl = this.getRequestUrl(endpoint, {}, { ...{}, ...options });
      const result: SyncAnalyzeResponse = await this.post(requestUrl, body).then((result) =>
        result.json()
      );

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }

  /**
   * Analyzes a text-based audio source synchronously.
   *
   * @param source - The text-based audio source to analyze.
   * @param options - Optional analysis options.
   * @param endpoint - The API endpoint to use for the analysis. Defaults to ":version/read".
   * @returns A promise that resolves to the analysis response, or an error if the analysis fails.
   */
  async analyzeText(
    source: TextSource,
    options?: AnalyzeSchema,
    endpoint = ":version/read"
  ): Promise<DeepgramResponse<SyncAnalyzeResponse>> {
    try {
      let body;

      if (isTextSource(source)) {
        body = JSON.stringify(source);
      } else {
        throw new DeepgramError("Unknown source type");
      }

      if (options !== undefined && "callback" in options) {
        throw new DeepgramError(
          "Callback cannot be provided as an option to a synchronous requests. Use `analyzeUrlCallback` or `analyzeTextCallback` instead."
        );
      }

      const requestUrl = this.getRequestUrl(endpoint, {}, { ...{}, ...options });
      const result: SyncAnalyzeResponse = await this.post(requestUrl, body).then((result) =>
        result.json()
      );

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }

  /**
   * Analyzes a URL-based audio source asynchronously.
   *
   * @param source - The URL-based audio source to analyze.
   * @param callback - The URL to call back with the analysis results.
   * @param options - Optional analysis options.
   * @param endpoint - The API endpoint to use for the analysis. Defaults to ":version/read".
   * @returns A promise that resolves to the analysis response, or an error if the analysis fails.
   */
  async analyzeUrlCallback(
    source: UrlSource,
    callback: CallbackUrl,
    options?: AnalyzeSchema,
    endpoint = ":version/read"
  ): Promise<DeepgramResponse<AsyncAnalyzeResponse>> {
    try {
      let body;

      if (isUrlSource(source)) {
        body = JSON.stringify(source);
      } else {
        throw new DeepgramError("Unknown source type");
      }

      const requestUrl = this.getRequestUrl(
        endpoint,
        {},
        { ...options, callback: callback.toString() }
      );
      const result: AsyncAnalyzeResponse = await this.post(requestUrl, body).then((result) =>
        result.json()
      );

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }

  /**
   * Analyzes a text-based audio source asynchronously.
   *
   * @param source - The text-based audio source to analyze.
   * @param callback - The URL to call back with the analysis results.
   * @param options - Optional analysis options.
   * @param endpoint - The API endpoint to use for the analysis. Defaults to ":version/read".
   * @returns A promise that resolves to the analysis response, or an error if the analysis fails.
   */
  async analyzeTextCallback(
    source: TextSource,
    callback: CallbackUrl,
    options?: AnalyzeSchema,
    endpoint = ":version/read"
  ): Promise<DeepgramResponse<AsyncAnalyzeResponse>> {
    try {
      let body;

      if (isTextSource(source)) {
        body = JSON.stringify(source);
      } else {
        throw new DeepgramError("Unknown source type");
      }

      const requestUrl = this.getRequestUrl(
        endpoint,
        {},
        { ...options, callback: callback.toString() }
      );
      const result: AsyncAnalyzeResponse = await this.post(requestUrl, body, {
        headers: { "Content-Type": "deepgram/audio+video" },
      }).then((result) => result.json());

      return { result, error: null };
    } catch (error) {
      if (isDeepgramError(error)) {
        return { result: null, error };
      }

      throw error;
    }
  }
}

export { ReadRestClient as ReadClient };
