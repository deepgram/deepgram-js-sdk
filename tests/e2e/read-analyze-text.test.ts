import { createClient } from "../../src/index";
import { ReadRestClient } from "../../src/core/packages/ReadRestClient";
import { DeepgramError } from "../../src/core/lib/errors";
import { structureOnlySerializer, setupApiMocks, cleanupApiMocks } from "../__utils__";
import { testReadSources, commonAnalysisOptions } from "../__fixtures__/read";

describe("read analyzeText E2E", () => {
  let deepgram: ReturnType<typeof createClient>;

  beforeAll(() => {
    // Set up API mocks (only active when not updating snapshots)
    setupApiMocks();

    // Use mock API key when mocking, real one when updating snapshots
    const apiKey = process.env.DEEPGRAM_API_KEY || "mock-api-key";
    deepgram = createClient(apiKey);

    // Add our custom serializer
    expect.addSnapshotSerializer(structureOnlySerializer);
  });

  afterAll(() => {
    // Clean up mocks
    cleanupApiMocks();
  });

  it("should analyze text synchronously", async () => {
    const { result, error } = await deepgram.read.analyzeText(
      testReadSources.simpleText,
      commonAnalysisOptions
    );

    // Verify no error occurred
    expect(error).toBeNull();
    expect(result).toBeTruthy();

    // Type guard to ensure result is not null for subsequent operations
    if (!result) {
      throw new Error("Result should not be null after toBeTruthy check");
    }

    // Test the structure with snapshot
    expect(result).toMatchSnapshot("read-analyzeText-response-structure");

    // Essential structural validation
    expect(result).toHaveProperty("metadata");
    expect(result).toHaveProperty("results");

    // Verify metadata structure
    expect(result.metadata).toHaveProperty("request_id");
    expect(result.metadata).toHaveProperty("created");
    expect(typeof result.metadata.request_id).toBe("string");
    expect(typeof result.metadata.created).toBe("string");

    // Verify results structure based on requested analysis options
    if (commonAnalysisOptions.sentiment) {
      expect(result.results).toHaveProperty("sentiments");
      if (result.results.sentiments) {
        expect(result.results.sentiments).toHaveProperty("segments");
        expect(Array.isArray(result.results.sentiments.segments)).toBe(true);
      }
    }

    if (commonAnalysisOptions.intents) {
      expect(result.results).toHaveProperty("intents");
      if (result.results.intents) {
        expect(result.results.intents).toHaveProperty("segments");
        expect(Array.isArray(result.results.intents.segments)).toBe(true);
      }
    }

    if (commonAnalysisOptions.summarize) {
      expect(result.results).toHaveProperty("summary");
    }
  }, 30000);

  it("should analyze text with different analysis options", async () => {
    const sentimentOnlyOptions = {
      sentiment: true,
      language: "en",
    };

    const { result, error } = await deepgram.read.analyzeText(
      testReadSources.sentimentText,
      sentimentOnlyOptions
    );

    expect(error).toBeNull();
    expect(result).toBeTruthy();

    if (!result) {
      throw new Error("Result should not be null after toBeTruthy check");
    }

    expect(result).toHaveProperty("results");

    // Should have sentiment analysis since it was requested
    expect(result.results).toHaveProperty("sentiments");

    // Verify sentiment structure
    const sentiments = result.results.sentiments;
    if (sentiments) {
      expect(sentiments).toHaveProperty("segments");
      expect(Array.isArray(sentiments.segments)).toBe(true);

      if (sentiments.segments.length > 0) {
        const segment = sentiments.segments[0];
        expect(segment).toHaveProperty("sentiment");
        expect(segment).toHaveProperty("sentiment_score");
        expect(typeof segment.sentiment).toBe("string");
        expect(typeof segment.sentiment_score).toBe("number");
      }
    }
  }, 30000);

  it("should reject synchronous analysis with callback option", async () => {
    const optionsWithCallback = {
      ...commonAnalysisOptions,
      callback: "https://example.com/callback",
    };

    const { result, error } = await deepgram.read.analyzeText(
      testReadSources.simpleText,
      optionsWithCallback
    );

    // Should return an error, not throw
    expect(result).toBeNull();
    expect(error).toBeTruthy();
    expect(error?.message).toContain("Callback cannot be provided as an option to a synchronous");
  }, 30000);

  it("should handle DeepgramError and return it in response", async () => {
    const apiKey = process.env.DEEPGRAM_API_KEY || "mock-api-key";
    const readClient = new ReadRestClient({ key: apiKey });

    // Mock the post method to throw a DeepgramError
    const mockError = new DeepgramError("Test DeepgramError");
    jest.spyOn(readClient as any, "post").mockRejectedValue(mockError);

    const { result, error } = await readClient.analyzeText(
      testReadSources.simpleText,
      commonAnalysisOptions
    );

    expect(result).toBeNull();
    expect(error).toBe(mockError);
    expect(error).toBeInstanceOf(DeepgramError);

    // Clean up spy
    jest.restoreAllMocks();
  });

  it("should re-throw non-DeepgramError", async () => {
    const apiKey = process.env.DEEPGRAM_API_KEY || "mock-api-key";
    const readClient = new ReadRestClient({ key: apiKey });

    // Mock the post method to throw a regular Error
    const mockError = new Error("Network error");
    jest.spyOn(readClient as any, "post").mockRejectedValue(mockError);

    await expect(
      readClient.analyzeText(testReadSources.simpleText, commonAnalysisOptions)
    ).rejects.toThrow("Network error");
    await expect(
      readClient.analyzeText(testReadSources.simpleText, commonAnalysisOptions)
    ).rejects.toThrow(Error);

    // Clean up spy
    jest.restoreAllMocks();
  });
});
