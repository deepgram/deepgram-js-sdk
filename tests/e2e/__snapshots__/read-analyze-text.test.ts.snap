// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`read analyzeText E2E should analyze text synchronously: read-analyzeText-response-structure 1`] = `
{
  "metadata": {
    "created": "<string>",
    "intents": {
      "model_metadata": {},
      "model_name": "<string>",
      "model_uuid": "<string>"
    },
    "language": "<string>",
    "request_id": "<string>",
    "sentiment": {
      "model_metadata": {},
      "model_name": "<string>",
      "model_uuid": "<string>"
    },
    "summary": {
      "model_metadata": {},
      "model_name": "<string>",
      "model_uuid": "<string>"
    }
  },
  "results": {
    "intents": {
      "segments": [
        {
          "end_word": "<number>",
          "intents": [
            {
              "confidence": "<number>",
              "intent": "<string>"
            }
          ],
          "start_word": "<number>",
          "text": "<string>"
        }
      ]
    },
    "sentiments": {
      "average": {
        "sentiment": "<string>",
        "sentiment_score": "<number>"
      },
      "segments": [
        {
          "end_word": "<number>",
          "sentiment": "<string>",
          "sentiment_score": "<number>",
          "start_word": "<number>",
          "text": "<string>"
        }
      ]
    },
    "summary": {
      "short": "<string>"
    }
  }
}
`;
